### ISSUE TITLE
[Bug] weird spikes in bucket - collecting feedback.

### ISSUE THREAD

**Issue by jeroenterheerdt** [jeroenterheerdt/HAsmartirrigation#35]
**Describe the bug**
We have seen situations where suddenly the daily adjusted run time bucket is jumping up for no good reason. Please let us know if you see this as well and under what circumstances.

**Comment by Justahobby01** [comment]
Yes, since I installed the integration, about two weeks ago, I have noticed it. I have been trying to figure out if it is occurring  after I restart HA.

**Comment by jeroenterheerdt** [comment]
it is occurring at restarts, but I cannot figure out why. I must be missing something. Next step is to add rigorous logging everywhere and see what is going on.

**Comment by Justahobby01** [comment]
I assumed it was my settings since I just took a guess at the flow rate and sq ft of my yard, but I set everything up correctly last night and saw the spike again today. I can add a debug to my logger and see if anything pops up.

**Comment by jeroenterheerdt** [comment]
added lots of logging in v0.0.45. please have a look at the logs

**Comment by Justahobby01** [comment]
I rebooted the host and restarted HA about ten times yesterday but didn't see a spike while the bucket was empty. Today, the bucket was at 108 seconds but spiked to 2692 sec after a restart. Debug is below:

[irrigation debug.txt](https://github.com/jeroenterheerdt/HAsmartirrigation/files/4794211/irrigation.debug.txt)

**Comment by jeroenterheerdt** [comment]
thanks for sharing this @Justahobby01 - the values you refer to are for adjusted run time, not for the bucket, but I was able to find them. The part where the problem occurs is this:
```
2020-06-17 12:21:28 INFO (MainThread) [custom_components.smart_irrigation.sensor] async_added_t_hass type: Daily Adjusted Run Time state: 108, attributes: {'adjusted_run_time_minutes': 1.8, **'bucket': '-0.17'**, 'force_mode_duration': 0, 'force_mode_enabled': False, 'friendly_name': 'Daily Adjusted Run Time', 'icon': 'mdi:sprinkler', 'lead_time': 0, 'maximum_duration': -1, 'unit_of_measurement': 's', 'water_budget': 91.57}
2020-06-17 12:21:28 INFO (MainThread) [custom_components.smart_irrigation.sensor] async_added_to_hass restoring state, **settting bucket to: -109.67719298065954**
```
The bucket is erroneously set to -109.67... but as you can see the bucket value discovered to restore is -0.17, which seems much more normal. As a result is bucket = -109 your water budget and daily adjusted run time go sky-high.
I will see if I can understand how this bucket value of -109 came up.

**Comment by jeroenterheerdt** [comment]
I have found a bug! not sure if we have now solved the problem, but at least it looks better on my side now. Please continue monitoring it @Justahobby01, you're a big help! I appreciate it!

**Comment by Justahobby01** [comment]
Awesome! I'm honored to be of assistance. What was the bug, if you don't mind me asking? 

I am using imperial and use smartweather for everything except minimum and maximum temps since I didn't see where smartweather provided that info.

**Comment by Justahobby01** [comment]
I changed the refresh time to get new values and then restarted HA.......and the values stayed the same. Looks promising!

**Comment by jeroenterheerdt** [comment]
ok, so we're in the same boat, I am on imperial as well as using smart weather where I can (but OWM for min and max temps). Figured I could always create a min and max temperature sensor in HA some time. Not for now though :).

Fingers crossed it works correctly now, keep me posted. Also, note that in v0.0.48 I added the `set_bucket` service which allows you to set the bucket to a specific value instead of just setting it to 0 with `reset_bucket`.
